iceoryx_hoofs/test/moduletests/test_posix_file.cpp

```diff
@@ -0,0 +1,486 @@
+                   .create(m_sut_file_path);
+    ASSERT_FALSE(sut.has_error());
+
+    auto result = sut->write(test_content.data(), test_content.size());
+    ASSERT_FALSE(result.has_error());
+    EXPECT_THAT(*result, Eq(test_content.size()));
+
+    result = sut->write_at(4, test_content.data(), test_content.size());
+    ASSERT_FALSE(result.has_error());
+    EXPECT_THAT(*result, Eq(test_content.size()));
+
+    std::array<uint8_t, 10> read_content{0};
+    auto read = sut->read(read_content.data(), read_content.size());
+
+    ASSERT_FALSE(read.has_error());
+    EXPECT_THAT(*read, Eq(6));
+    EXPECT_THAT(read_content, Eq(std::array<uint8_t, 10>{240, 250, 0, 0, 240, 250, 0, 0, 0, 0}));
+}
+
+TEST_F(File_test, MoveConstructedFileWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1f9ffd26-2171-4a1c-880b-c28ed11c843b");
+
+    const std::array<uint8_t, 2> test_content{242, 252};
+    auto sut = FileBuilder()
+                   .open_mode(OpenMode::PURGE_AND_CREATE)
+                   .permissions(perms::owner_read | perms::owner_write)
+                   .access_mode(posix::AccessMode::READ_WRITE)
+                   .create(m_sut_file_path);
+    ASSERT_FALSE(sut.has_error());
+
+    auto result = sut->write(test_content.data(), test_content.size());
+    ASSERT_FALSE(result.has_error());
+    EXPECT_THAT(*result, Eq(test_content.size()));
+
+    File sut2{std::move(sut.value())};
+
+    std::array<uint8_t, 3> read_content{0};
+    auto read = sut2.read(read_content.data(), read_content.size());
+
+    ASSERT_FALSE(read.has_error());
+    EXPECT_THAT(*read, Eq(2));
+    EXPECT_THAT(read_content, Eq(std::array<uint8_t, 3>{242, 252, 0}));
+}
+
+TEST_F(File_test, MoveAssignedFileWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "70e19f33-5007-42be-8a5e-51df6ca429dd");
+
+    const std::array<uint8_t, 2> test_content{244, 254};
+    auto sut = FileBuilder()
+                   .open_mode(OpenMode::PURGE_AND_CREATE)
+                   .permissions(perms::owner_read | perms::owner_write)
+                   .access_mode(posix::AccessMode::READ_WRITE)
+                   .create(m_sut_file_path);
+    ASSERT_FALSE(sut.has_error());
+
+    auto result = sut->write(test_content.data(), test_content.size());
+    ASSERT_FALSE(result.has_error());
+    EXPECT_THAT(*result, Eq(test_content.size()));
+
+    File sut2{std::move(sut.value())};
+    auto sut3 = FileBuilder()
+                    .open_mode(OpenMode::OPEN_EXISTING)
+                    .permissions(perms::owner_read)
+                    .access_mode(posix::AccessMode::READ_ONLY)
+                    .create(m_sut_file_path);
+    ASSERT_FALSE(sut3.has_error());
+
+    sut2 = std::move(sut3.value());
+
+    std::array<uint8_t, 3> read_content{0};
+    auto read = sut2.read(read_content.data(), read_content.size());
+
+    ASSERT_FALSE(read.has_error());
+    EXPECT_THAT(*read, Eq(2));
+    EXPECT_THAT(read_content, Eq(std::array<uint8_t, 3>{244, 254, 0}));
```

Comment:
Can you check what `get_file_handle` returns? Otherwise, it would also be possible that the move assignment does nothing; the test would still succeed (if I'm not mistaken).